### ISSUE TITLE
Framework golden tests are failing on 3.16 stable infra

### ISSUE THREAD

**Issue by CaseyHillers** [flutter/flutter#139673]
Example failure: https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20web_canvaskit_tests_6/11802/overview

This started from https://github.com/flutter/flutter/pull/139162, which seems expected. What's unexpected is the tests are failing.

**Comment by CaseyHillers** [comment]
@Piinks I think there's a few possible paths here:

1. I should accept these goldens. However, I worry that might break HEAD. If it does, we may need some extra fixes to the gold infra to support our release branches.
2. Skia gold shouldn't run on release branches

Do you know if either of those are the preferred option here?

**Comment by Piinks** [comment]
I thought gold was ignored on release branches? I am not familiar with the CI on stable or why this would be happening

**Comment by Piinks** [comment]
https://flutter-gold.skia.org/ only tracks the main branch

**Comment by Piinks** [comment]
> I should accept these goldens. However, I worry that might break HEAD. If it does, we may need some extra fixes to the gold infra to support our release branches.

Is there an option to approve them somewhere? This is rather mystifying. The release branch has never tracked goldens

**Comment by CaseyHillers** [comment]
In the logs, I'm seeing

```
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure was thrown running a test:
Expected: one widget whose rasterized image matches golden image
"m3.navigation_bar.default.indicator.inkwell.shape.png"
  Actual: _TypeWidgetFinder:<Found 1 widget with type "NavigationBar": [
            NavigationBar(dependencies: [_InheritedTheme, _LocalizationsScope-[GlobalKey#f5898]]),
          ]>
   Which: SkiaException: Skia Gold received an unapproved image in post-submit
          testing. Golden file images in flutter/flutter are triaged
          in pre-submit during code review for the given PR.

          Visit https://flutter-gold.skia.org// to view and approve
          the image(s), or revert the associated change. For more
          information, visit the wiki:
          https://github.com/flutter/flutter/wiki/Writing-a-golden-file-test-for-package:flutter

          Debug information for Gold --------------------------------
          stdout: Given image with hash 38fe9a59b6a997c999d905a314af7f53 for test
material.m3.navigation_bar.default.indicator.inkwell.shape
          Expectation for test: 655d132cb4e791ff24b2abbfb7fa9c93 (positive)
          Expectation for test: abaec32e87402d343fc89fee20e43322 (positive)
          Expectation for test: b33997d5d439c1d12b22b9a95f1ba2ce (positive)
          Expectation for test: baa7d985fc9d065f6e8aad07da51d2d7 (positive)
          Expectation for test: 1a8dc17f1fd43d4d9290a85fc721a3c4 (positive)
          Untriaged or negative image:
https://flutter-gold.skia.org/detail?grouping=name%3Dmaterial.m3.navigation_bar.default.indicator.inkwell.shape%26source_type%3Dflutter&digest=38fe9a59b6a997c999d905a314af7f53


          stderr: Test: material.m3.navigation_bar.default.indicator.inkwell.shape FAIL


          result-state.json: No result file found.
```

Which links to https://flutter-gold.skia.org/detail?grouping=name%3Dmaterial.m3.navigation_bar.default.indicator.inkwell.shape%26source_type%3Dflutter&digest=38fe9a59b6a997c999d905a314af7f53
 as where to go for triage.

**Comment by Piinks** [comment]
Wow, how strange it generated a digest when the main page does not show there are unapproved images. I think we should approve it to unblock the tree for now.
I am looking to see how we disabled gold on release branches, since it is only happening on one test shard I think it probably fell through the cracks

**Comment by Piinks** [comment]
I found this so far https://github.com/flutter/flutter/issues/58814, but that was handled in flutter/cocoon. I am thiinking this is likely to do with the config in ci.yaml

**Comment by Piinks** [comment]
Ah ok I think I have found why this is happening on the single shard. There must have been a change in canvaskit that is not yet on stable, because the canvas kit image above is different from the one on main. So when the CP landed, it generated a new canavaskit image because stable does not have the canvaskit change. This is a gap in the 'gold ignores release branches' coverage since CPs typically generate already approved images. I am approving the iamge now to unblock the tree.

**Comment by Piinks** [comment]
I think I've run into a separate issue. It looks like Gold will not allow me to approve the image since it is not off the main branch. So the stable branch tree is stuck until either the canvaskit change rolls in or we patch in something that will make that test non-failing.

@kjlubick am I right thinking if an unapproved image reaches Gold and it is not on the main branch, we can't check that image in?

@CaseyHillers WDYT of me pushing a change that ignores Gold results on `candidate` branches that we could then CP into the stable branch?

**Comment by Piinks** [comment]
Or I could update the ci.yaml to split out the gold shards using `enabled_branches:`/`disabled_branches:` to really restrict gold testing to only the master branch.

**Comment by kjlubick** [comment]
>  am I right thinking if an unapproved image reaches Gold and it is not on the main branch, we can't check that image in?

Images can be uploaded to either "the primary branch" or a "secondary branch". Secondary branches are (currently) exclusively pull requests, but they were named that way to support the idea of a branch off the primary one and have a separate batch of images. However, this is not fully implemented.

Right now if you upload metadata and images to Gold and tell it "This is on the primary branch with commit `abc123`", Gold will reject that data if the commit is not actually on the main branch (and thus the images never get ingested, which means they cannot be triaged).

You could also upload meatadata and images to Gold and say "Hey, this is with PR `stable`", to which Gold (I think) will currently reject because such a PR with that id does not exist. In the future, this may work, but not yet.

**Comment by CaseyHillers** [comment]
> @CaseyHillers WDYT of me pushing a change that ignores Gold results on candidate branches that we could then CP into the stable branch?

SGTM!
